### PR TITLE
Added tests for validate on search.py

### PR DESCRIPTION
**Pull request by thebigG**
# Tests for validate method on search.py

## Description
Hi all,

Hope you are all doing well.

Noticed that we were not testing the validate method on `search.py`, so I added all those tests.

## Context of change

Please add options that are relevant and mark any boxes that apply.

- [x] Software (software that runs on the PC)
- [ ] Library (library that runs on the PC)
- [ ] Tool (tool that assists coding development)
- [ ] Other

## Type of change

Please mark any boxes that apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.

- [x] Tested the test suite as a whole on `Ubuntu 20`
- [x] Tested `test_search.py` in isolation on `Ubuntu 20`

## Checklist:

Please mark any boxes that have been completed.

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: tests/config/test_search.py [hunk 1]
```diff
@@ -60,3 +60,96 @@ def test_search_config_init(mocker, locale, domain, exp_domain):
...
         # Assertions
         assert cfg.domain == exp_domain
+
+
+@pytest.mark.parametrize('keywords, province_or_state, locale, providers, in_city',
+                         [(['Ice Cream', 'Spiderman'], None, Locale.USA_ENGLISH, [enums.Provider.INDEED], 'Austin')])
+def test_search_config_validate_invalid_province(keywords, province_or_state, locale, providers, in_city):
+    cfg = SearchConfig(keywords, province_or_state, locale, providers, city=in_city)
+
+    with pytest.raises(AssertionError) as e:
```
Comment (PaulMcInnis) [inline tests/config/test_search.py:70]: you can actually do this as `pytest.raises(AssertionError, match=r"Province/State not set")`

**Review by PaulMcInnis — CHANGES_REQUESTED** [review]
Looks good, lets swap over the string assertions to using the built-in `match` keyword

**Review by markkvdb — COMMENTED** [review]
Nice job! Seems good to check what kind of search configurations should be valid. 

Looking at the file it seems that we can further abstract these tests. In essence, we provide a search configuration and check if JobFunnel raises an AssertionError with a certain message. This leads me to think that we could potentially define a list of search configurations and expected error message. After we just need one test function, say, `test_search_config_validate_invalid`, and provide the list of search configurations.

**Comment by thebigG** [comment]
> Nice job! Seems good to check what kind of search configurations should be valid.
> 
> Looking at the file it seems that we can further abstract these tests. In essence, we provide a search configuration and check if JobFunnel raises an AssertionError with a certain message. This leads me to think that we could potentially define a list of search configurations and expected error message. After we just need one test function, say, `test_search_config_validate_invalid`, and provide the list of search configurations.

@markkvdb I'm always hesitant to doing this when writing tests. Because usually if you pass a list with all  permutations of configurations to a single test, you might have to have `if` statements in there. And that can mess with the readability of what should be a very simple test. Though I know sometimes that can be done in a somewhat clean way, but in this case I suspect we will have logic that might add  unnecessary complexity to what could be a very simple test.

I try my best to keep tests simple and dumb, especially unit tests. If the tests are simple and dumb, points of failure are very easy to investigate. 

I know this approach can be very slow, but I hope the reasoning behind it makes sense.

File: tests/config/test_search.py (hunk 1 above)
Comment (thebigG) [reply tests/config/test_search.py]: Has been added on 0a9dd3102db59ce200b304a2e463301ab2e6d616

**Review by PaulMcInnis — APPROVED** [review]
🚢 

@markkvdb I think theres room for more functional testing as well in the future, but this is a good coverage expansion for our unit tests at a lower level. 

Thx @thebigG

**Pull request merged**